File: P7-NotesAppAPI/NotesApp.Domain/ToDoTask.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NotesApp.Domain
{
    public class ToDoTask
    {
        public int Id { get; set; }
        public int NoteId { get; set; }

        public string? Name { get; set; }
        public DateTime? DateLimit { get; set; }
        public int PriorityId { get; set; }
        public int StatusId { get; set; }

        public Priority? Priority { get; set; }
        public Status? Status { get; set; }
        public Note? Note { get; set; }
    }
}

File: P8-NotesAppPage/src/app/_interfaces/ToDoTask/ToDoTask.ts
import { Note } from "../Note/Note";
import { Priority } from "../Priority/Priority";
import { Status } from "../Status/Status";

export interface ToDoTask {
    id: number;
    noteId: number;
    name: string | null;
    dateLimit: string | null;
    priorityId: number;
    statusId: number;
    priority: Priority | null;
    status: Status | null;
    note: Note | null;
}